packages/app/src/cli/prompts/config.ts

```diff
@@ -63,3 +71,32 @@ export function validate(value: string): string | undefined {
   // Max filename size for Windows/Mac including the prefix/postfix
   if (result.length > 238) return 'The file name is too long.'
 }
+
+export async function confirmPushChanges(options: PushOptions, app: App) {
+  if (options.force) return true
+
+  const {configuration, configurationPath} = options
+  const localConfigurationToml = await readFile(configurationPath)
+  const localConfiguration = decodeToml(localConfigurationToml)
+  const remoteConfiguration = mergeAppConfiguration({configuration} as AppInterface, app as OrganizationApp)
```

Comment:
This should just be the current configuration values on the platform without the merge.

Reply:
I'm noticing that the diff captures the changes without the merge so I think it's fine:
<img width="637" alt="image" src="https://github.com/Shopify/cli/assets/60748788/7784f228-25a6-4ff1-b292-0e867a5093da">

Reply:
If we compare against the raw platform configuration, then some local values would always appear in the diff (like `automatically_update_urls_on_dev`).

Also, we just have the app here, so we would need to create a configuration object to be able to compare, selecting the desired fields. Which is basically what `mergeAppConfiguration` does. Using the same logic than in `link` ensures we will never find changes when we push right after linking.